Formula/k/kubectl-cnpg.rb

```diff
@@ -0,0 +1,31 @@
+class KubectlCnpg < Formula
+  desc "CloudNativePG plugin for kubectl"
+  homepage "https://cloudnative-pg.io/"
+  url "https://github.com/cloudnative-pg/cloudnative-pg/archive/refs/tags/v1.23.1.tar.gz"
+  sha256 "c13373baf2a0d177d82f85be66b963ff6d55853aaafc7cbbe9afa904d75b8695"
+  license "Apache-2.0"
+  head "https://github.com/cloudnative-pg/cloudnative-pg.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildVersion=#{version}
+      -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildCommit=#{tap.user}
```

Comment:
I am not sure if it is possible to get the actual commit hash here? `#{Utils.git_head}` doesn't work since I use a tar.gz ball, but at the end of the day, the commit hash is probably not that important.

Other formular also just use `tap.user` in thier build step:
https://github.com/Homebrew/homebrew-core/blob/4b8f468391dfae5de4e8be919976e10391f034de/Formula/v/vet.rb#L23

Reply:
Yes, this is fine! We can use a Git checkout to use `Utils.git_head`, but Git checkouts are more expensive than tarballs.

I'd go with a Git checkout only if the output of `kubectl-cnpg version` makes it look like `"#{tap.user}"` is meant to be a Git hash when it's not (e.g. it says something like `commit: Homebrew`, which doesn't really make sense).

Reply:
I'll change it to a Git Checkout. Version currently shows Homebrew:

```bash
kubectl cnpg version
Build: {Version:1.23.1 Commit:Homebrew Date:2024-04-30T09:21:30Z}
```